app/app.go

```diff
@@ -414,6 +415,27 @@ func New(
 	)
 	app.SetEndBlocker(app.EndBlocker)
 
+	planName := "v2.0.0"
+	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx sdk.Context, _ upgradetypes.Plan) {
+		// https://github.com/crypto-org-chain/chain-main/blob/master/doc/architecture/adr-003.md
+	})
+
+	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(err)
+	}
+
+	if upgradeInfo.Name == planName && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
```

Comment:
could add some explain for IsSkipHeight,
such as height > upgradeInfo.Height?

Reply:
in latest planName is different from "2.0.0"? 
and not fall into this step?

Reply:
cosmos-sdk has a feature that you can skip upgrades in command line:
```
$ chain-maind start --help
...
      --unsafe-skip-upgrades ints                       Skip a set of upgrade heights to continue the old binary
```